src/main/webapp/dev/js/main/instructorCourseEnrollPage.js

```diff
@@ -42,8 +42,118 @@ if (window.addEventListener) {
     window.attachEvent('load', loadUpFunction);
 }
 
+const container = document.getElementById('spreadsheet');
+
+/**
+ * Holds Handsontable settings, reference and other information for Spreadsheet.
+ */
+/* global Handsontable:false */
+const handsontable = new Handsontable(container, {
+    manualColumnResize: true,
+    manualRowResize: true,
+    manualColumnMove: true,
+    rowHeaders: true,
+    colHeaders: ['Section', 'Team', 'Name', 'Email', 'Comments'],
+    columnSorting: true,
+    className: 'htCenter',
+    sortIndicator: true,
+    maxCols: 5,
+    maxRows: 100,
+    stretchH: 'all',
+    minSpareRows: 1,
+    contextMenu: [
+        'row_above',
+        'row_below',
+        'remove_row',
+        'undo',
+        'redo',
+        'make_read_only',
+        'alignment',
+    ],
+});
+
+/**
+ * Updates column header order and generates a header string.
+ *
+ * Example: Change this array ['Section', 'Team', 'Name', 'Email', 'Comments']
+ * into a string = "Section | Team | Name | Email | Comments"
+ */
+function updateHeaderOrder() {
+    const colHeaders = handsontable.getColHeader().join('|');
+    return colHeaders.concat('\n');
+}
+
+/**
+ * Updates the student data from the spreadsheet when any of the
+ * following event occurs afterChange, afterColumnMove or afterRemoveRow.
+ *
+ * Pushes the output data into the textarea (used for form submission).
+ *
+ */
+function updateDataDump() {
+    const spreadsheetData = handsontable.getData();
+    let dataPushToTextarea = updateHeaderOrder();
+
+    for (let row = 0; row < spreadsheetData.length; row += 1) {
+        let countEmptyColumns = 0;
+        let rowData = '';
+        const spreadsheetDataRow = spreadsheetData[row];
+
+        for (let col = 0; col < spreadsheetDataRow.length; col += 1) {
+            rowData += spreadsheetDataRow[col] || '';
+            if (col < spreadsheetDataRow.length - 1) {
+                rowData += '|';
+                if (!spreadsheetDataRow[col]) {
+                    countEmptyColumns += 1;
+                }
+            }
+        }
+
+        if (countEmptyColumns < spreadsheetData[row].length - 1) {
+            dataPushToTextarea += rowData.concat('\n');
+        }
+    }
+
+    $('#enrollstudents').text(dataPushToTextarea);
+}
+
+/**
+ * Adds the listener to specified hook name and only for this Handsontable instance.
+ */
+function addHandsontableHooks() {
```

Comment:
An alternative approach is to update the `textarea` only when the user click submit before actually submitting the form. The browser will  need to do less work. What do you think?

Reply:
I agree with this :) Initially this was done to facilitate easier debugging through the browser developer tools. 

Will be attaching an id to the form and trying something like this to update the data before actual submission:
`$('#student-data-spreadsheet-form').submit(() => updateDataDump());`